cores/esp8266/core_esp8266_features.h

```diff
@@ -33,7 +33,30 @@
 #define WIFI_HAS_EVENT_CALLBACK
 
 
+#ifndef __STRINGIFY
+#define __STRINGIFY(a) #a
+#endif
 
+// these low level routines provide a replacement for SREG interrupt save that AVR uses
+// but are esp8266 specific. A normal use pattern is like
+//
+//{
+//    uint32_t savedPS = xt_rsil(1); // this routine will allow level 2 and above
+//    // do work here
+//    xt_wsr_ps(savedPS); // restore the state
+//}
+//
+// level (0-15), interrupts of the given level and above will be active
+// level 15 will disable ALL interrupts,
+// level 0 will enable ALL interrupts,
+//
+#define xt_rsil(level) (__extension__({uint32_t state; __asm__ __volatile__("rsil %0," __STRINGIFY(level) : "=a" (state) :: "memory"); state;}))

```

Comment:
About, host tests or emulation on host, I think this is sufficient around ps state and cycle count:
```
#ifndef CORE_MOCK
...
#endif // not CORE_MOCK
```